driver-core/src/main/java/com/datastax/driver/core/Cluster.java

```diff
@@ -2208,16 +2208,35 @@ public void ensurePoolsSizing() {
         }
 
         public PreparedStatement addPrepared(PreparedStatement stmt) {
```

Comment:
Indeed, but at least the new version would not print a log warning about re-preparing the same statement, if it's being re-prepared because the resultset metadata has changed.